noir/postcheckout.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+# Post-checkout hook for warning the user to create a patch file before running a build,
+# if there are outstanding commits that could get lost if we switch to a different noir-repo checkout.
+set -euo pipefail
+
+cd $(dirname $0)
+
+is_branch=$3
+
+if [ "$is_branch" == "0" ]; then
+    exit 0
+fi
+
+needs_patch=$(scripts/sync.sh needs-patch)
```

Comment:
more typical would be needs-patch has a good exit code only if patch needed then doing
`if scripts/sync.sh needs-patch; then`